src/main/kotlin/org/magmaoffenburg/roboviz/gui/MainWindow.kt

```diff
@@ -0,0 +1,171 @@
+        size = Dimension(Graphics.frameWidth, Graphics.frameHeight)
+        jMenuBar = JMenuBar()
+        iconImage = ImageIO.read(MainWindow::class.java.getResource("/images/icon.png"))
+
+        // set the main window position
+        location = if (Graphics.centerFrame) {
+            SwingUtils.centerWindowOnScreen(this, Point(0,0))
+        } else {
+            Point(Graphics.frameX, Graphics.frameY)
+        }
+        toFront()
+
+        addWindowListener(object : WindowAdapter() {
+            override fun windowClosing(e: WindowEvent?) {
+                Graphics.frameWidth = width
+                Graphics.frameHeight = height
+
+                Main.config.write()
+                Renderer.instance.dispose(glCanvas)
+            }
+        })
+    }
+
+    private fun initializeMenu() {
+        jMenuBar.removeAll()
+
+        when (Main.mode) {
+            DataTypes.Mode.LIVE -> initializeLiveMenu()
+            DataTypes.Mode.LOG -> initializeLogMenu()
+        }
+    }
+
+    private fun initializeLiveMenu() {
+        val connection = ConnectionMenu(this)
+        val server = ServerMenu()
+        val view = ViewMenu()
+        val camera = CameraMenu()
+        val help = HelpMenu()
+
+        // Live Mode specific items
+        view.addSeparator()
+        view.addItem("Toggle Server Speed", KeyEvent.VK_M) { println("TODO toggleShowServerSpeed") }
+        view.addItem("Playmode Overlay", KeyEvent.VK_O) { println("TODO openPlaymodeOverlay") }
+        view.addItem("Log Mode", KeyEvent.VK_F4) { Main.changeMode() }
+
+        // add connection menu to config change listeners
+        Main.config.addConfigChangedListener(connection)
+
+        jMenuBar.add(connection)
+        jMenuBar.add(server)
+        jMenuBar.add(view)
+        jMenuBar.add(camera)
+        jMenuBar.add(help)
+    }
+
+    private fun initializeLogMenu() {
+        val view = ViewMenu()
+        val camera = CameraMenu()
+        val help = HelpMenu()
+
+        view.addSeparator()
+        view.addItem("Live Mode", KeyEvent.VK_F4) { Main.changeMode() }
+
+        jMenuBar.add(view)
+        jMenuBar.add(camera)
+        jMenuBar.add(help)
+    }
+
+    private fun initializeGLCanvas() {
+        val glProfile = GLProfile.get(GLProfile.GL2)
+        val glCapabilities = GLCapabilities(glProfile)
+
+        // set glCapabilities according to config
+        glCapabilities.stereo = Graphics.useStereo
+        if (Graphics.useFsaa) {
+            glCapabilities.sampleBuffers = true
+            glCapabilities.numSamples = Graphics.fsaaSamples
+        }
+
+        glCanvas = GLCanvas(glCapabilities)
+        glCanvas.focusTraversalKeysEnabled = false
+
+        add(glCanvas, BorderLayout.CENTER)
+    }
+
+    private fun initializeLogPlayerControls() {
+        logPlayerControls = LogPlayerControlsPanel()
+        logPlayerControls!!.isVisible = true
+    }
+
+    // TODO investigate window title for logMode
```

Comment:
Yes, i don't think changing the window title is useful, since RoboVis can toggle between Log and Live Mode. But I have yet to come with a replacement for it.

Reply:
resolved in a0a7b1a by @hannesbraun